File: 2144-maximum-difference-between-increasing-elements/2144-maximum-difference-between-increasing-elements.cs
public class Solution {
    public int MaximumDifference(int[] nums) {
        int result = -1;
        int minDiff = nums[0];
        for(int i = 1; i < nums.Length;i++){
            if(nums[i] > minDiff){
                result = Math.Max(result, nums[i] - minDiff);
            }
            else{
                minDiff = nums[i];
            }
        }
        return result;
    }
}

File: 2144-maximum-difference-between-increasing-elements/2144-maximum-difference-between-increasing-elements.ts
function maximumDifference(nums: number[]): number {
    let minDiff = nums[0];
    let result = -1;
    for(var i = 0; i < nums.length; i++){
        if(nums[i] > minDiff){
            result = Math.max(result, nums[i] - minDiff);
        }
        else{
            minDiff = nums[i];
        }
    }
    return result;
};